radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/PeerManager.java

```diff
@@ -152,6 +156,61 @@ public PeerManager(
 		}
 	}
 
+	public CompletableFuture<PeerChannel> findOrCreateProxyChannel(NodeId nodeId) {
+		synchronized (lock) {
+			final var maybeActiveChannel = this.proxyCertificateManager.get().getVerifiedProxiesForNode(nodeId)
```

Comment:
Method body is somewhat too long, as for me. Perhaps moving part of the logic into dedicated methods would be helpful.

Reply:
I agree, I'll refactor it